core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -3187,6 +3187,20 @@ private void convertJoin(Blackboard bb, SqlJoin join) {
     convertFrom(rightBlackboard, right);
     final RelNode tempRightRel = requireNonNull(rightBlackboard.root, "rightBlackboard.root");
 
+
+    // Append tracking of any added sub query nodes in either side of a join
+    for (int offset: leftBlackboard.offsetNodes) {
+      fromBlackboard.offsetNodes.add(offset);
+    }
+
+    final LookupContext left_rels = new LookupContext(
```

Comment:
Can you explain this line since the `left_rels` object is only used to compute a size?

Reply:
Nope, will replace with just the line getting the size